src/curvefit/solvers/solvers.py

```
@@ -99,4 +100,32 @@ def fit(self, data, x_init=None, options=None):
                 xs_opt.append(self.solver.x_opt)
             
             self.x_opt = xs_opt[np.argmin(fun_vals)]
-            self.fun_val_opt = np.min(fun_vals)
\ No newline at end of file
+            self.fun_val_opt = np.min(fun_vals)
+
+
+class SmartInitialization(Composite):
+
+    def __init__(self, update_prior_fun):
+        self.update_prior_fun = update_prior_fun
+
+    def fit(self, data, x_init=None, options=None):
+         if self.is_solver_defined():
```

Should we raise some error if solver is not defined?